lib/Command/Room/TRoomCommand.php

```diff
@@ -369,8 +369,13 @@ protected function completeParticipantValues(CompletionContext $context): array
 			return [];
 		}
 
-		return array_map(function (Participant $participant) {
-			return $participant->getUser();
-		}, $room->searchParticipants($context->getCurrentWord()));
+		$users = [];
+		foreach ($room->searchParticipants($context->getCurrentWord()) as $participant) {
+			if (!$participant->isGuest()) {
+				$users[] = $participant->getUser();
+			}
+		}
+
+		return $users;
```

Comment:
return array_map(function (Participant $participant) {
```suggestion
return array_filter(array_map(function (Participant $participant) {
			return $participant->getUser();
		}, $room->searchParticipants($context->getCurrentWord())));
```

should work too?

Reply:
It indeed should work, too. But that would be `O(2)` complexity compared to `O(1)`. It's not like it makes much of a difference, I also feel like `foreach` is more readable than `array_filter()` + `array_map()`. But if you want it changed I'll change it. Just let me know :smiley: